examples/face_recognition_11/face_recognition_11/evaluator.py

```diff
@@ -0,0 +1,234 @@
+
+    thresholds = [compute_threshold(zip(ground_truths, inferences), fmr) for fmr in baseline_fmr_x]
+
+    n_genuine_pairs = np.sum([gt.is_same for gt in ground_truths])
+    n_imposter_pairs = np.sum([not gt.is_same for gt in ground_truths])
+
+    fnmr_y = list()
+    fmr_y = list()
+
+    for threshold in thresholds:
+        tsm_for_one_threshold = [compute_per_sample(gt, inf, threshold) for gt, inf in zip(ground_truths, inferences)]
+        n_fm = np.sum([metric.is_false_match and not metric.failure_to_enroll for metric in tsm_for_one_threshold])
+        n_fnm = np.sum(
+            [metric.is_false_non_match and not metric.failure_to_enroll for metric in tsm_for_one_threshold],
+        )
+        fnmr_y.append((n_fnm / n_genuine_pairs) * 100)
+        fmr_y.append(n_fm / n_imposter_pairs)
+
+    curve_test_case_fnmr = CurvePlot(
+        title="Test Case FNMR vs. Baseline FMR",
+        x_label="Baseline False Match Rate",
+        y_label="Test Case False Non-Match Rate (%)",
+        x_config=AxisConfig(type="log"),
+        curves=[Curve(x=baseline_fmr_x, y=fnmr_y, extra=dict(Threshold=thresholds))],
+    )
+
+    curve_test_case_fmr = CurvePlot(
+        title="Test Case FMR vs. Baseline FMR",
+        x_label="Baseline False Match Rate",
+        y_label="Test Case False Match Rate",
+        x_config=AxisConfig(type="log"),
+        y_config=AxisConfig(type="log"),
+        curves=[Curve(x=baseline_fmr_x, y=fmr_y, extra=dict(Threshold=thresholds))],
+    )
+
+    genuine_values = [
+        inf.similarity for gt, inf in zip(ground_truths, inferences) if gt.is_same and inf.similarity is not None
+    ]
+
+    imposter_values = [
+        inf.similarity for gt, inf in zip(ground_truths, inferences) if not gt.is_same and inf.similarity is not None
+    ]
+
+    min_data = min(min(genuine_values), min(imposter_values))
+    max_data = max(max(genuine_values), max(imposter_values))
+
+    number_of_bins = 50
+    bin_size = (max_data - min_data) / number_of_bins
+    bin_edges = [min_data + i * bin_size for i in range(number_of_bins + 1)]
+
+    hist1_adjusted = list(np.histogram(genuine_values, bins=bin_edges, density=True)[0])
+    hist2_adjusted = list(np.histogram(imposter_values, bins=bin_edges, density=True)[0])
+
+    # histogram of the relative distribution of genuine and imposter pairs, bucketed by similarity score.
+    similarity_dist = Histogram(
+        title="Similarity Distribution",
+        x_label="Similarity Score",
+        y_label="Frequency (%)",
+        buckets=list(bin_edges),
+        frequency=list([hist1_adjusted, hist2_adjusted]),
+        labels=["Genuine Pairs", "Imposter Pairs"],
+    )
+
+    return [similarity_dist, curve_test_case_fnmr, curve_test_case_fmr]
+
+
+def compute_test_suite_metrics(
+    test_case_metrics: List[Tuple[TestCase, TestCaseMetrics]],
+    threshold: float,
+) -> TestSuiteMetrics:
+    baseline: TestCaseMetrics = test_case_metrics[0][1]
```

Comment:
How do we know the first test case is the baseline? I would rather not make this assumption